tasks/icinga2-RedHat.yml

```diff
@@ -2,5 +2,15 @@
 
 - name: RedHat - Add Icinga 2 repository
   become: yes
-  template: src=ICINGA-release.repo dest=/etc/yum.repos.d/ICINGA-release.repo
-  when: i2_use_repository
+  template:
+    src: ICINGA-release.repo
```

Comment:
Where is the template file ?
Templates files need to have the .j2 extension at the end 

also - for consistency, if you are using apt_repository, it is also makes send to use yum_repository 
https://docs.ansible.com/ansible/latest/modules/yum_repository_module.html